tests/admin_utils/tests.py

```diff
@@ -329,12 +329,27 @@ def test_callable(obj):
 
         self.assertEqual(label_for_field(lambda x: "nothing", Article), "--")
         self.assertEqual(label_for_field("site_id", Article), "Site id")
-        # The correct name and attr are returned when `__` is in the field
-        # name.
-        self.assertEqual(label_for_field("site__domain", Article), "Site  domain")
+
+        # Related lookup, field doesn't have a explicit verbose_name defined
+        domain_field_attr_name = "domain"
+        domain_field = Site._meta.get_field(domain_field_attr_name)
+        self.assertIsNone(domain_field._verbose_name)
+        self.assertEqual(
+            label_for_field("site__domain", Article), domain_field_attr_name
+        )
         self.assertEqual(
             label_for_field("site__domain", Article, return_attr=True),
-            ("Site  domain", Site._meta.get_field("domain")),
+            ("domain", domain_field),
+        )
+
+        # Related lookup, field does have a explicit verbose_name defined
+        owner_verbose_name = "Site owner"
+        owner_field = Site._meta.get_field("owner")
+        self.assertEqual(owner_field.verbose_name, owner_verbose_name)
+        self.assertEqual(label_for_field("site__owner", Article), owner_verbose_name)
+        self.assertEqual(
+            label_for_field("site__owner", Article, return_attr=True),
+            (owner_verbose_name, owner_field),
         )
```

Comment:
I can see that you are trying to be explicit about these being verbose names but I would be tempted to simplify this to:
```python
        # Related lookup, field doesn't have an explicit verbose_name defined.
        self.assertEqual(
            label_for_field("site__domain", Article), "Domain"
        )
        # Related lookup, field has an explicit verbose_name defined.
        self.assertEqual(label_for_field("site__owner", Article), "Site owner")
```

Reply:
Right, pushing this on `65ec1d1: fix: smaller test case`